### ISSUE TITLE
Studio Form Panel creates Invalid OAuth 2 Security Definitions

### ISSUE THREAD

**Issue by philsturgeon** [stoplightio/studio#226]
**Describe the bug**

When creating oauth2 Security Definitions for OAS2 using Studio via the Form Panel, it will use the structure from OAS3 instead, which invalidates the document. 

**To Reproduce**

Create a new OAS2 YAML document, and click "Security Schemes +".

<img width="1069" alt="Screen Shot 2019-10-29 at 16 59 27" src="https://user-images.githubusercontent.com/67381/67808699-761b0a00-fa97-11e9-810b-b732f732804e.png">
<img width="672" alt="Screen Shot 2019-10-29 at 16 59 31" src="https://user-images.githubusercontent.com/67381/67808700-76b3a080-fa97-11e9-8b36-9c3f373e6dfd.png">

This will populate the Write panel with the following:

```yaml
securityDefinitions:
  API Key - 1:
    type: oauth2
    flows:
      implicit:
        authorizationUrl: ''
        refreshUrl: ''
        scopes: []
      password:
        tokenUrl: ''
        scopes: []
      clientCredentials:
        tokenUrl: ''
        refreshUrl: ''
        scopes: []
      authorizationCode:
        authorizationUrl: ''
        tokenUrl: ''
        refreshUrl: ''
        scopes: []
```

1. OAS2 does not have `flows`, only `flow`. It is not possible for one securityDefinition to cover multiple flows, it would instead need to be multiple definitions. 

```yaml
securityDefinitions:
  oauth2Implicit:
    type: oauth2
    flow: implicit
    authorizationUrl: 'http://example.org/oauth'

  oauth2Password:
    type: oauth2
    flow: password
    authorizationUrl: 'http://example.org/oauth'
```

The properties that were nested inside "password", "implicit", etc. must be siblings.

2. There is no `refreshUrl` in OAS2

<img width="945" alt="image" src="https://user-images.githubusercontent.com/67381/67809246-a7480a00-fa98-11e9-91ad-4f041c035bff.png">

3. Studio is trying to create Scopes as an array, but they should instead be an object with a description:

```yaml
# bad 
type: oauth2
authorizationUrl: http://swagger.io/api/oauth/dialog
flow: implicit
scopes:
  - write:pets
  - read:pets

# good
type: oauth2
authorizationUrl: http://swagger.io/api/oauth/dialog
flow: implicit
scopes:
  write:pets: modify pets in your account
  read:pets: read your pets
```

4. Studio Forms has no way to make these descriptions in the current UI.

This issue is just for "Forms" creating valid things. There is an existing story for Docs: https://github.com/stoplightio/studio/issues/108

**Comment by philsturgeon** [comment]
So far Studio Form and Docs support looks a bit like this: 

<img width="982" alt="image" src="https://user-images.githubusercontent.com/67381/67810017-15d99780-fa9a-11e9-8607-fa1482282698.png">

**Comment by philsturgeon** [comment]
Closed in v1.6.0.
